pulp_rpm/tests/functional/api/test_domains.py

```diff
@@ -0,0 +1,533 @@
+            pulp_domain=domain2.name, repository=repo2, set_id=True
+        )
+        assert advisory2.id == an_id
+
+        assert advisory1.pulp_href != advisory2.pulp_href
+    finally:
+        cleanup_domains([domain1, domain2], cleanup_repositories=True)
+
+
+@pytest.mark.parallel
+def test_cross_domain_copy_all(
+    monitor_task,
+    rpm_copy_api,
+    cleanup_domains,
+    setup_domain,
+):
+    """Test attempting to copy between different Domains."""
+    domain1 = None
+    domain2 = None
+    try:
+        domain1, remote1, src1, dest1 = setup_domain()
+        domain2, remote2, src2, dest2 = setup_domain()
+        # Success, everything in domain1
+        data = Copy(
+            config=[
+                {"source_repo_version": src1.latest_version_href, "dest_repo": dest1.pulp_href}
+            ],
+            dependency_solving=False,
+        )
+        monitor_task(rpm_copy_api.copy_content(data, pulp_domain=domain1.name).task)
+
+        # Failure, call and src domain1, dest domain2
+        with pytest.raises(ApiException):
+            data = Copy(
+                config=[
+                    {"source_repo_version": src1.latest_version_href, "dest_repo": dest2.pulp_href}
+                ],
+                dependency_solving=False,
+            )
+            monitor_task(rpm_copy_api.copy_content(data, pulp_domain=domain1.name).task)
+
+        # Failure, call domain2, src/dest domain1
+        with pytest.raises(ApiException):
+            data = Copy(
+                config=[
+                    {"source_repo_version": src1.latest_version_href, "dest_repo": dest1.pulp_href}
+                ],
+                dependency_solving=False,
+            )
+            monitor_task(rpm_copy_api.copy_content(data, pulp_domain=domain2.name).task)
+
+    finally:
+        cleanup_domains([domain1, domain2], cleanup_repositories=True)
+
+
+@pytest.mark.parallel
+def test_cross_domain_content(
+    cleanup_domains,
+    monitor_task,
+    rpm_advisory_api,
+    rpm_copy_api,
+    rpm_repository_api,
+    setup_domain,
+):
+    """Test the content parameter."""
+    domain1 = None
+    domain2 = None
+    try:
+        domain1, remote1, src1, dest1 = setup_domain()
+        domain2, remote2, src2, dest2 = setup_domain()
+
+        # Copy content1 from src1 to dest1, expect 2 copied advisories
+        advisories1 = rpm_advisory_api.list(
+            repository_version=src1.latest_version_href, pulp_domain=domain1.name
+        ).results
+        advisories_to_copy1 = (advisories1[0].pulp_href, advisories1[1].pulp_href)
+
+        data = Copy(
+            config=[
+                {
+                    "source_repo_version": src1.latest_version_href,
+                    "dest_repo": dest1.pulp_href,
+                    "content": advisories_to_copy1,
+                }
+            ],
+            dependency_solving=False,
+        )
+        monitor_task(rpm_copy_api.copy_content(data, pulp_domain=domain1.name).task)
```

Comment:
Same here in this test, `monitor_task` isn't necessary as the task should never get launched.

Reply:
My general approach to these negative tests is to "do the normal thing", with an expect-error around it. Do we have-to monitor the task? No. But monitoring task-returns is the normal path for code like this. Especially since people use the tests as examples of "how to client", I prefer this.

Reply:
I agree. Also in case the task is erroneously launched, this will ensure proper cleanup of resources afterwards.